launchpad-guaranteed-tickets-v2/src/token_release.rs

```diff
@@ -66,7 +66,6 @@ pub trait TokenReleaseModule:
     #[only_owner]
     #[endpoint(setUnlockSchedule)]
     fn set_unlock_schedule(&self, unlock_milestones: MultiValueEncoded<MultiValue2<u64, u64>>) {
-        self.require_add_tickets_period();
```

Comment:
Please do a test where you have an unlock schedule like 25/25/50, a user claims the first one, then you update the unlock schedule to 50/25/25, user claims again. I'm really not sure this works properly.

For reference, user should have 25% claimed first, then +25%, i.e. 50% total claimed at the end.

Reply:
Yes, I understand what you're saying, but the contract is still not exploitable or has any real issues. The problem here would be that the users that claimed their tokens early on, got to have more tokens than they should according to the second vesting logic. But that's normal and cannot be avoided. 
Anyway this PR will not be merged to main, as this is not the desired default behaviour of this SC.

Reply:
One extra check was added to avoid any negative deduction.

Reply:
Added the test for this case.